flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/TestingAbstractInvokables.java

```diff
@@ -82,4 +84,51 @@ public void invoke() throws Exception {
 			}
 		}
 	}
+
+	public static final class TestInvokableRecordCancel extends AbstractInvokable {
+
+		private static final Object lock = new Object();
+		private static CompletableFuture<Boolean> gotCanceledFuture = new CompletableFuture<>();
```

Comment:
It would be easier to use a static CompletableFuture here since the object is not initialized in the test code and it's hard to get a reference to the CompletableFuture if it is not static.